### ISSUE TITLE
Can't get exception reporting to work on OSX

### ISSUE THREAD

**Issue by koenbok** [getsentry/raven-objc#54]
No matter what I try, I can't get exceptions to log on OSX. I can verify it installs the handler with `NSSetUncaughtExceptionHandler`. But if I throw an exception after a few seconds, the handler never gets called. Logging seems to work fine.

**Comment by asbjornu** [comment]
Are you using the `sharedClient` as in #53? If not, please provide all relevant code so it's possible to make an informed opinion on why it isn't working.

**Comment by koenbok** [comment]
Yes I do. I created a simple example project. It throws an exception after 3 seconds, but the custom exception handler never gets called.

https://www.dropbox.com/s/ess0zhshf7bdilt/RavenTest.zip?dl=0

**Comment by asbjornu** [comment]
Are you setting the `sharedClient` as described in #53?

**Comment by koenbok** [comment]
Yes.

```
- (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
    
    // Just a test Sentry project
    RavenClient *client = [RavenClient clientWithDSN:@"https://6df0670f7c6e443cb2c18030b7ea7b19:6a19c63382524319a0d6ea31ac8275ea@app.getsentry.com/41381"];
    
    [RavenClient setSharedClient:client];
    [[RavenClient sharedClient] setupExceptionHandler];
    
    NSLog(@"Exception handler installed");
    
    [self performSelector:@selector(crash) withObject:nil afterDelay:3];
    
}
```

**Comment by asbjornu** [comment]
Since you're crashing the whole application, the exception won't be sent until you restart the application, as noted by the following sentence in the readme:
> Note: when using the global exception handler, exceptions will be sent the next time the app is started.

Is that relevant information in any way?

**Comment by koenbok** [comment]
Nope. I understand what needs to happen but the handler just never get's called. Not on the crash, not on the next launch.

**Comment by asbjornu** [comment]
Perhaps @timorzadir has any ideas?

**Comment by timorzadir** [comment]
I've tried all the released pods from 0.3.0 to 1.0.1 and I can't get global exception reporting to call the handler either. 

Further investigation shows that the NSSetUncaughtExceptionHandler that the RavenClient uses behind the scenes, doesn't actually work the same way in OSX as in iOS. 

Currently I belive that global exception handling is a missing feature in the client for OSX.

**Comment by koenbok** [comment]
I came to that conclusion as well.

**Comment by joerick** [comment]
This is a little bit kooky (maybe not appropriate for a library to include), but this category on NSApplication makes OS X apps call the handler like I think they should.

```objc
@interface NSApplication (MXException)

- (void)reportException:(NSException *)theException;

@end

@implementation NSApplication (MXException)

- (void)reportException:(NSException *)exception
{
    (*NSGetUncaughtExceptionHandler())(exception);
}

@end
```
